### PR TITLE
Deprecate execute()

### PR DESCRIPTION
**Pull request by ajavadia**
### Summary
The execute() function has been around for a while but it is more of a script rather than part of the SDK.
Originally it was a replacement for 3 steps: transpile, assemble, run. Now it is a substitute for just two steps (transpile and run), so the convenience is diluted. It has also not kept up with recent transpile or run options (e.g. https://github.com/Qiskit/qiskit-terra/issues/7640).
A proper high-level interface is being developed as part of Qiskit Primitives, which allows for much richer tasks such as calculation of expectation values and error mitigation. 

In this commit I'm just adding a `PendingDeprecationWarning` to start to migrate user behavior.

### CODE REVIEW

**Comment by ajavadia** [comment]
> `execute()` reminds me a lot of Django's [`render()` function](https://docs.djangoproject.com/en/4.0/topics/http/shortcuts/#render) (which lives in `django.shortcuts`). Sure, one needs only a few lines of code to create a template and then render it to an HTTP response, but this pattern is so incredibly common that it's helpful to have a convenience function for it. This function is in widespread use throughout Django projects and is typically used any time additional control is not necessary.
> 
> I don't understand the future direction of Primitives enough to know whether the analogy with Django holds, but if the pattern of "transpile & run" without additional options will continue to be common, then I think there is value in having a convenience function for it.

I very much agree with convenient ways of doing things. The problem with this particular function is that it goes against the direction of Qiskit runtimes and primitives. The recently added Sampler should basically provide the same thing for users who want to execute circuits (plus more since you can also do some error mitigation on probability distributions).

File: releasenotes/notes/deprecate-execute-67a5c68b93722057.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+   For example, instead of running::
+
+      from qiskit import execute
+      job = execute(circuit, backend)
+
+   you can run::
+
+      from qiskit import transpile
+      new_circuit = transpile(circuit)
+      job = backend.run(new_circuit)
```
Comment (1ucian0) [inline releasenotes/notes/deprecate-execute-67a5c68b93722057.yaml:20]:
```suggestion
      from qiskit import transpile
      new_circuit = transpile(circuit, backend)
      job = backend.run(new_circuit)
```

**Comment by garrison** [comment]
> A proper high-level interface is being developed as part of Qiskit Primitives, which allows for much richer tasks such as calculation of expectation values and error mitigation.

> The recently added Sampler should basically provide the same thing for users who want to execute circuits (plus more since you can also do some error mitigation on probability distributions).

As this PR stands currently, the deprecation warning suggests migrating to `transpile()` followed by `backend.run()` (with the primitives mentioned as an alternative), while the added release note does not mention primitives at all.  If the end goal is for users to migrate to the Sampler primitive, I believe it would be best to emphasize this in both places and helpful to provide users in both places with suggested code for doing so.  In my opinion, providing users with a clear migration path at the time of the deprecation is even more important than doing the deprecation in as early a release as possible.  (That said, I appreciate that this PR landed early, so that there is plenty of time to discuss before the next release.)

**Comment by 1ucian0** [comment]
I rewrote this PR in https://github.com/Qiskit/qiskit/pull/11044

**Comment by 1ucian0** [comment]
Closing in favor of https://github.com/Qiskit/qiskit/pull/11044

**Pull request closed without merging**